ibm/service/power/resource_ibm_pi_instance_action.go

```diff
@@ -0,0 +1,213 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+	"fmt"
+
+	st "github.com/IBM-Cloud/power-go-client/clients/instance"
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"log"
+	"time"
+)
+
+func ResourceIBMPIInstanceAction() *schema.Resource {
+	return &schema.Resource{
+
+		CreateContext: resourceIBMPIInstanceActionCreate,
+		ReadContext:   resourceIBMPIInstanceActionRead,
+		UpdateContext: resourceIBMPIInstanceActionUpdate,
+		DeleteContext: resourceIBMPIInstanceActionDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Update: schema.DefaultTimeout(15 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PI Cloud instnce id",
+			},
+			Arg_PVMInstanceId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PVM Instance ID",
+			},
+			Arg_PVMInstanceActionType: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"start", "stop", "hard-reboot", "soft-reboot", "immediate-shutdown"}),
+				Description:  "PVM instance action type",
+			},
+			Arg_PVMInstanceHealthStatus: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{PVMInstanceHealthOk, PVMInstanceHealthWarning}),
+				Default:      "OK",
+				Description:  "Set the health status of the PVM instance to connect it faster",
+			},
+
+			// Computed
+			Attr_Status: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the PVM instance",
+			},
+			Attr_Progress: {
+				Type:        schema.TypeFloat,
+				Computed:    true,
+				Description: "The progress of an operation",
```

Comment:
Should the comment say action instead of operation?

Reply:
I have copied it from swagger so lets keep it.